### DISCUSSION TITLE
'Operation not supported' when writing to block device

### DISCUSSION THREAD

**Discussion by DylanVanAssche** [fwupd/fwupd#4988] [Show and tell]
**Describe the question**

I copied some parts of the uf2 plugin to write to a block device.
However, when I start writing, fwupdmgr returns 'Operation not supported'.
I grepped through the fwupd sources but I couldn't find where this message is returned.
Is the write firmware method in the uf2 plugin not suitable for this?

**fwupd version information**
Please provide the version of the daemon and client if applicable to your current installation of `fwupd`.

1.8.4

Source: https://github.com/DylanVanAssche/fwupd/tree/dd-plugin

**Comment by hughsie** [comment]
I've added some comments to your branch; does that help?
